finality-aleph/src/party.rs

```diff
@@ -200,8 +200,8 @@ async fn run_aggregator<B, C, BE>(
                         //This is only for optimization purposes.
                         continue;
                     }
-                    for new_hash in batch {
-                        let to_finalize_headers = chain_extension_step(last_finalized, new_hash, client.as_ref());
+                    for new_block_data in batch {
+                        let to_finalize_headers = chain_extension_step(last_finalized, new_block_data.0, client.as_ref());
```

Comment:
`chain_extension_step` also needs some minor changes, as it assumes that `new_block_data.0` has been imported which might not be the case. The current code probably works fine, but it logs an error in case this rare case happens -- the log message should be adjusted, and the level should be `warn` the way I see it...

Also, in the current version this probably does not work if someone inputs a block hash with an incorrect block number -- make sure this is handled.

Reply:
A test for the latter would be also great. The behavior should be that if someone inputs (hash, fake_num), and `num(hash) != fake_num` then such an entry should be ignored.

Reply:
I think the simplest would be to take `AlephData` in `chain_extension_step` (instead of a hash) and implement appropriate logic there.